build.rs

```diff
@@ -1,95 +0,0 @@
-use std::{env, path::PathBuf, process::Command};
-
-use which::which;
-
-/* Uncomment for debugging, you can't print normally from a build.rs
-macro_rules! bprintln {
-    ($($arg:tt)*) => {{
-        println!("cargo:warning={}", format!($($arg)*))
-    }}
-}
-*/
-
-macro_rules! berr {
-    ($($arg:tt)*) => {{
-        Error::Err { msg: format!($($arg)*) }
-    }}
-}
-
-fn main() {
-    match run() {
-        Ok(()) => {}
-        Err(e) => {
-            panic!("build failed: {}", e);
-        }
-    }
-}
-
-// build the overlay .so file
-fn run() -> Result<(), Error> {
-    println!("cargo:rerun-if-changed=src/env_shim.c");
-    // println!("cargo:rerun-if-changed=src/pam_motd_overlay_versions.ldscript");
-
-    let out_dir = env::var("OUT_DIR").map_err(|e| berr!("no OUT_DIR: {}", e))?;
-
-    let mut target_so = PathBuf::from(out_dir);
-    target_so.push("env_shim.so");
-
-    let cc = select_cc()?;
```

Comment:
FYI https://docs.rs/cc/latest/cc/

AFAIK that's the defacto standard way of building c source files in build.rs

Reply:
Yeah that was the first thing I tried (not for the particular crate the the first time I did some cursed LD_PRELOAD shennaigans in rust as part of the motd crate), but unfortunately it does not support producing .so files. I think it only produces .a files.